### PR TITLE
Update support batch delete storage design document 

### PR DESCRIPTION
**Pull request by ZhangYu0123**
## Proposed changes

The detail delete column design in storage level for batch delete function feature #4051 .
We discuss the concrete delete column structure and update the design document here.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on #ISSUE, and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

File: docs/en/internal/doris_storage_optimization.md [hunk 1]
```diff
@@ -164,17 +197,18 @@ message ColumnMetaPB {
...
 }
 
-message FileFooterPB {
+message SegmentFooterPB {
 	optional uint32 version = 2 [default = 1]; // 用于版本兼容和升级使用
 	repeated ColumnPB schema = 5; // 列Schema
-    optional uint64 num_values = 4; // 文件中保存的行数
-    optional uint64 index_footprint = 7; // 索引大小
-    optional uint64 data_footprint = 8; // 数据大小
+  optional uint64 num_values = 4; // 文件中保存的行数
```
Comment (EmmyMiao87) [inline docs/en/internal/doris_storage_optimization.md:203]: Please use English comment

File: docs/en/internal/doris_storage_optimization.md [hunk 2]
```diff
@@ -164,17 +197,18 @@ message ColumnMetaPB {
...
+  optional uint64 index_footprint = 7; // 索引大小
+  optional uint64 data_footprint = 8; // 数据大小
 	optional uint64 raw_data_footprint = 8; // 原始数据大小
 
-    optional CompressKind compress_kind = 9 [default = COMPRESS_LZO]; // 压缩方式
-    repeated ColumnMetaPB column_metas = 10; // 列元数据
+  optional CompressKind compress_kind = 9 [default = COMPRESS_LZO]; // 压缩方式
+  repeated ColumnMetaPB column_metas = 10; // 列元数据
 	optional PagePointerPB key_index_page; // short key索引page
+  optional PagePointerPB delete_index_page; // delete index索引page
```
Comment (EmmyMiao87) [inline docs/en/internal/doris_storage_optimization.md:211]: Same as above

File: docs/en/internal/doris_storage_optimization.md (hunk 1 above)
Comment (ZhangYu0123) [reply docs/en/internal/doris_storage_optimization.md]: done

File: docs/en/internal/doris_storage_optimization.md (hunk 2 above)
Comment (ZhangYu0123) [reply docs/en/internal/doris_storage_optimization.md]: dome

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by yangzhg — APPROVED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**